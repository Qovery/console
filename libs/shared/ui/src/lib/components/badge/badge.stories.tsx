import type { Meta } from '@storybook/react'
import { Badge } from './badge'

const Story: Meta<typeof Badge> = {
  component: Badge,
  title: 'Badge',
  decorators: [
    (Story) => (
      <div style={{ background: 'white', padding: '3em' }}>
        <Story />
      </div>
    ),
  ],
}

export const Primary = {
  args: {
    children: 'Foobar',
  },
}

export default Story
