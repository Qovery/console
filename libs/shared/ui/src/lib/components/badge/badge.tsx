import { type VariantProps, cva } from 'class-variance-authority'
import { type ComponentPropsWithoutRef, type ElementRef, forwardRef } from 'react'
import { twMerge } from '@qovery/shared/util-js'

const badgeVariants = cva(['inline-flex', 'items-center', 'shrink-0', 'font-medium', 'text-xs', 'h-6', 'px-2'], {
  variants: {
    color: {
      neutral: [],
      red: ['text-red-500'],
      purple: ['text-purple-500'],
      sky: ['text-sky-500'],
      green: ['text-green-500'],
      yellow: ['text-yellow-500'],
      brand: ['text-brand-500'],
    },
    variant: {
      outline: ['border'],
      surface: ['border'],
    },
    radius: {
      rounded: ['rounded'],
      full: ['rounded-full'],
    },
  },
  compoundVariants: [
    {
      variant: 'surface',
      color: 'neutral',
      className: ['bg-neutral-100', 'border-neutral-200', 'text-neutral-400'],
    },
    {
      variant: 'outline',
      color: 'neutral',
      className: ['border-neutral-250', 'text-neutral-350', 'dark:border-neutral-350', 'dark:text-neutral-300'],
    },
    /*
    // Generate all colors
      ...[
        'red' as const,
        'purple' as const,
        'sky' as const,
        'green' as const,
        'yellow' as const,
        'brand' as const,
      ].flatMap((color) => [
        {
          variant: 'surface' as const,
          color,
          className: [`bg-${color}-50`],
        },
        {
          variant: ['surface' as const, 'outline' as const],
          color,
          className: [`border-${color}-500`],
        },
      ]),
  */
    {
      variant: 'surface',
      color: 'red',
      className: ['bg-red-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'red',
      className: ['border-red-500'],
    },
    {
      variant: 'surface',
      color: 'purple',
      className: ['bg-purple-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'purple',
      className: ['border-purple-500'],
    },
    {
      variant: 'surface',
      color: 'sky',
      className: ['bg-sky-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'sky',
      className: ['border-sky-500'],
    },
    {
      variant: 'surface',
      color: 'green',
      className: ['bg-green-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'green',
      className: ['border-green-500'],
    },
    {
      variant: 'surface',
      color: 'yellow',
      className: ['bg-yellow-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'yellow',
      className: ['border-yellow-500'],
    },
    {
      variant: 'surface',
      color: 'brand',
      className: ['bg-brand-50'],
    },
    {
      variant: ['surface', 'outline'],
      color: 'brand',
      className: ['border-brand-500'],
    },
  ],
  defaultVariants: {
    variant: 'outline',
    color: 'neutral',
    radius: 'rounded',
  },
})

export interface BadgeProps
  extends Omit<ComponentPropsWithoutRef<'span'>, 'color'>,
    VariantProps<typeof badgeVariants> {}

export const Badge = forwardRef<ElementRef<'span'>, BadgeProps>(function Badge(
  { className, color, variant, radius, ...badgeProps },
  forwardedRef
) {
  return (
    <span
      data-accent-color={color}
      {...badgeProps}
      ref={forwardedRef}
      className={twMerge(badgeVariants({ color, variant, radius }), className)}
    />
  )
})

export default Badge
