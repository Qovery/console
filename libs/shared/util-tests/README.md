# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
