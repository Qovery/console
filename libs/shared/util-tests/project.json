{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
