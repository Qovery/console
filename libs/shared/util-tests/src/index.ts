export * from './lib/shared-util-tests'
