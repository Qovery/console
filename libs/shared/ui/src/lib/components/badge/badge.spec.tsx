import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import { Badge } from './badge'

describe('Badge', () => {
  it('shoud render children', () => {
    renderWithProviders(<Badge>Foobar</Badge>)
    screen.getByText('Foobar')
  })
  it('shoud match size snapshot', () => {
    const { baseElement } = renderWithProviders(<Badge size="md">Foobar</Badge>)
    expect(baseElement).toMatchSnapshot()
  })
  it('shoud match color snapshot', () => {
    const { baseElement } = renderWithProviders(<Badge color="green">Foobar</Badge>)
    expect(baseElement).toMatchSnapshot()
  })
  it('shoud match variant snapshot', () => {
    const { baseElement } = renderWithProviders(<Badge variant="surface">Foobar</Badge>)
    expect(baseElement).toMatchSnapshot()
  })
})
