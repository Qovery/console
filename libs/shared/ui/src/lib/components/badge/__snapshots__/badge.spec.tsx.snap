// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge shoud match color snapshot 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 text-green-500 border rounded border-green-500"
      data-accent-color="green"
    >
      Foobar
    </span>
  </div>
</body>
`;

exports[`Badge shoud match size snapshot 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded border-neutral-250 text-neutral-350 dark:border-neutral-350 dark:text-neutral-300"
    >
      Foobar
    </span>
  </div>
</body>
`;

exports[`Badge shoud match variant snapshot 1`] = `
<body>
  <div>
    <span
      class="inline-flex items-center shrink-0 font-medium text-xs h-6 px-2 border rounded bg-neutral-100 border-neutral-200 text-neutral-400"
    >
      Foobar
    </span>
  </div>
</body>
`;
